features/79-structured-logging/structured-logging.md

```diff
@@ -0,0 +1,69 @@
+# Structured Logging
+
+## Purpose
+
+To make log output structured so that the log output can be filtered or
+otherwise processed.
+
+### Log output
+
+### Common structured log data
+
+* process - process name
+* logger - name of the logger that generates the log entry
+* datetime - datetime when the log entry is emitted
+* level - indicate the level of severity for the log entry
+* message - log message
+
+### More log data
+
+The following items are not necessarily available in all situations, but
+they should be part of the structured log data if available.
+
+* request_id - request ID, generated by router
+
+### Logger
+
+Logger is the primary field for filtering log entries.
+
+Proposing the following list of loggers:
+
+* server - skygear-server log not otherwise covered by other loggers
+* server.sql - skygear-server log for executed sql statement and result
+* server.http - skygear-server log for http request and response body
+* server.transport skygear-server log for plugin transport
+* plugin - plugin log not otherwise covered by other loggers
+* plugin.transport - plugin log for plugin transport
+* cloudfunc - cloud function log generated by developer code
```

Comment:
I get the feeling that this category is to separate `developer code` from non `developer code`.